samples/testing-frameworks/appium/client-side/csharp/biometric-sample-test/BitbarBiometricSample/TestBiometricsIOS.cs

```diff
@@ -0,0 +1,83 @@
+        private RemoteWebDriver driver;
+
+        [OneTimeSetUp]
+        public void InitDriver()
+        {
+            const String AppiumHubURL = "https://appium.bitbar.com/wd/hub";
+
+            AppiumOptions capabilities = new();
+            //Platfrom dependent 
+            capabilities.AddAdditionalCapability("platformName", "iOS");
+            capabilities.AddAdditionalCapability("deviceName", "iPhone device");
+            capabilities.AddAdditionalCapability("automationName", "XCUITest");
+            //Customizable
+            capabilities.AddAdditionalCapability("bitbar_project", "C# Appium Automated Test");
+            capabilities.AddAdditionalCapability("bitbar_testrun", "iOS Run");
+            capabilities.AddAdditionalCapability("bitbar_device", "Apple iPhone 11");
+            //Required
+            capabilities.AddAdditionalCapability("bitbar_apiKey", "<YOUR_BITBAR_API_KEY>");
+            capabilities.AddAdditionalCapability("bitbar_app", "<APP_ID>");
+            //Biometrics capability
+            capabilities.AddAdditionalCapability("biometricInstrumentation", true);
+
+            driver = new IOSDriver<IOSElement>(new Uri(AppiumHubURL), capabilities, TimeSpan.FromSeconds(300));
+        }
+
+        [OneTimeTearDown]
+        public void QuitDriver()
+        {
+            driver.Quit();
+        }
+
+        [Test]
+        public void Test_Biometric_Authentication()
+        {
+            Console.WriteLine("Test execution started:");
+
+            //Accept initial biometric dialog
+            var wait = new WebDriverWait(driver, TimeSpan.FromSeconds(10)).Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementExists(By.XPath("//XCUIElementTypeAlert[@name=\"Biometric Authentication\"]")));
+            driver.ExecuteScript("mobile: alert", new Dictionary<string, string> { { "action", "accept" }, { "buttonLabel", "OK" } });
+
+            TakeScreenshot("HomeScreen");
+            var buttonBiometrics = driver.FindElement(By.XPath("(//*[@name=\"Biometric authentication\"])[3]"));
+            buttonBiometrics.Click();
+
+            var textSensorType = driver.FindElement(MobileBy.AccessibilityId("Sensor type value"));
+            Assert.That(textSensorType.Text.ToUpper, Is.EqualTo("TOUCHID"));
+
+            var buttonAskBiometrics = driver.FindElement(By.XPath("//XCUIElementTypeOther[@name=\"Ask biometric authentication\"]"));
+            buttonAskBiometrics.Click();
+            TakeScreenshot("Dialog");
+
+            driver.ExecuteScript("mobile: alert", new Dictionary<string, string> { { "action", "accept" }, { "buttonLabel", "Pass" } });
+
+            var textResult = driver.FindElement(MobileBy.AccessibilityId("Authentication status value"));
+            TakeScreenshot("Result");
+            Assert.That(textResult.Text.ToUpper, Is.EqualTo("SUCCEEDED"));
+        }
+
+        public void TakeScreenshot(String fileName)
```

Comment:
I think we should extract it to another class. That code is duplicated in the second test file

Reply:
Yes it is as I mentioned in the description but as it is a test sample made for template purposes I wanted to keep it simple as it can be without extracting single methods or capabilities to another files. Let me know what do you think about that approach.

Reply:
I believe extracting it may be beneficial for everyone. Users writing their own test may just copy class with utils like that screenshot to their own projects. It will be easier to change later or reuse in other test cases.